### PR TITLE
CI: add composite action for fork detection

### LINKED ISSUE #24737: CI: fork detection in setup-build composite action broken
## Description

As tested in https://github.com/camunda/camunda/pull/24676 [this check](https://github.com/camunda/camunda/blob/b81d9b9773c35628ddc67ba1fedf1dfbac199245/.github/actions/setup-build/action.yml#L104-L108) does not yield `true` on PRs from forked repositories because `github.event.name` is an empty string.

This leads to the Unified CI being broken in PRs from forked repositories which should only require a small fix to get working (it worked prior).

## Acceptance Tests

* create new composite action that encapsulates logic checking whether current pipeline runs in PR from forked repositories or in `camunda/camunda`
    * this allows to reuse the logic in many GHA workflows without duplication
* have a working logic in all contexts with correct results (in monorepo on PRs, on push, etc)
* refactor `setup-build` to use the new composite action
* add documentation to the action

## Hints

* check out https://github.com/orgs/community/discussions/25217 for how others solve this task

### PR DESCRIPTION
**Pull request by cmur2**
## Description

This PR introduces a new, small composite action that checks if current job runs on code from a forked repository (via PR). This information can be used to disable certain CI features that are not available on forks (e.g. access to secrets).

Why have a new composite action for a simple check like this? Rationale: a composite action allows reuse (predicted to affect many/all GHA workflows in this repository) and encapsulates the logic which might need updates later.

Before merging check that the action works correctly on:

- on PRs in `camunda/camunda` (all CI checks should run) ✔️ ([link](https://github.com/camunda/camunda/actions/runs/11934300012/job/33263175160?pr=24895))
- on PRs from forks (`setup-build` should not attempt to fetch secrets) ✔️ ([link](https://github.com/camunda/camunda/actions/runs/11952224899/job/33317563410?pr=24676))
- on merge queue (cannot be checked in `camunda/camunda`, demo repo acceptable) ✔️ ([link](https://github.com/camunda/infra-camunda24737-poc/actions/runs/11952652011/job/33318833437))
- on push ✔️ ([link](https://github.com/camunda/infra-camunda24737-poc/actions/runs/11952607180/job/33318692631))

## Checklist

- [x] for CI changes:
  - [x] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #24737

### CODE REVIEW

File: .github/actions/is-fork/action.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    description: Output whether the current job runs on code from a forked repository
+    value: ${{ steps.detect-fork.outputs.is-fork == 'true' }}
+
+runs:
+  using: composite
+  steps:
+  - id: detect-fork
+    shell: bash
+    run: |
+      echo is-fork="${{ github.event.pull_request.head.repo.fork }}" | tee -a $GITHUB_OUTPUT
```
Comment (cmur2) [inline .github/actions/is-fork/action.yml:22]: This value can be empty in some contexts (on push, in merge queue, on schedule) but is caught by the `== 'true'` comparison in line 14.

Comment (clementnero) [reply .github/actions/is-fork/action.yml:22]: Not blocking and even not problematic here, but I always use intermediate environment variable as a good practice for mitigating script injection attacks, as advised here https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions

**Review by clementnero — APPROVED** [review]
LGTM, thank you for the extended tests!

Only one non-blocking comment regarding intermediate environment variables.

File: .github/actions/is-fork/action.yml (hunk 1 above)
Comment (clementnero) [reply .github/actions/is-fork/action.yml:22]: Personal opinion: it also improves readability :)

Comment (cmur2) [reply .github/actions/is-fork/action.yml:22]: Yea I should do it as well.

**Pull request merged**